tensorflow/lite/java/demo/app/src/main/java/com/example/android/tflitecamerademo/ImageClassifier.java

```diff
@@ -176,12 +179,12 @@ public void useGpu() {
   }
 
   public void useCPU() {
-    tfliteOptions.setUseNNAPI(false);
     recreateInterpreter();
   }
 
   public void useNNAPI() {
-    tfliteOptions.setUseNNAPI(true);
+    nnapiDelegate = new NnApiDelegate();
+    tfliteOptions.addDelegate(nnapiDelegate);
     recreateInterpreter();
```

Comment:
Just confirming that you've tested this path on Android and it works?

Reply:
Yes, it works on various devices I tested. Some of them need extra setAllowFp16PrecisionForFp32() though.